addons/calendar_ical/models/res_users.py

```diff
@@ -0,0 +1,27 @@
+from odoo import models, _
+from odoo.addons.base.models.res_users import check_identity
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    @check_identity
+    def preference_calendar_setup_google(self):
+        return self._preference_calendar_setup('calendar_setup_google', _('Google Calendar'))
+
+    @check_identity
+    def preference_calendar_setup_iphone(self):
+        return self._preference_calendar_setup('calendar_setup_iphone', _('iPhone Calendar'))
+
+    def _preference_calendar_setup(self, tag, desc):
+        self.ensure_one()
+        base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
```

Comment:
> Line may contain deprecated content in future version: web.base.url. Please adapt changes or contact runbot team.

Not sure why Runbot doesn't like this line. I see dozens like it elsewhere in the codebase.

Reply:
Hi :wave: 
There shouldn't be a lot of occurrences left, we replaced those with https://github.com/odoo/odoo/pull/68201/commits/28da5b41b3c2d9a22770871d689809057d69d173
which was following https://github.com/odoo/odoo/pull/68201/commits/ae75cadd71bdd2a5599b0d741983c0d17f23d5a2
There were a few left since those where a bit tricky to change or didn't make sense to be changed, but most of the time it shouldn't be the case.

Since those commits, the runbot is warning about the deprecated use of web.base.url ICP instead of using the helper.

Reply:
In this case, you should probably use self.get_base_url() to get the website from the user if the user has one set, or the website from his company.
Anyway, if no website is returned, the get_base_url() will fallback to the ICP as last ressort.